noether_filtering/include/noether_filtering/cloud/impl/mls_smoothing_filter.hpp

```diff
@@ -0,0 +1,122 @@
+#ifndef NOETHER_FILTERING_CLOUD_IMPL_MLS_SMOOTHING_FILTER_HPP
+#define NOETHER_FILTERING_CLOUD_IMPL_MLS_SMOOTHING_FILTER_HPP
+
+#include "noether_filtering/cloud/mls_smoothing_filter.h"
+#include "noether_filtering/utils.h"
+#include <pcl/surface/mls.h>
+#include <pcl/search/kdtree.h>
+#include <console_bridge/console.h>
+#include <XmlRpcException.h>
+
+namespace noether_filtering
+{
+namespace cloud
+{
+template<typename PointT>
+const std::string MLSSmoothingFilter<PointT>::POLYNOMIAL_ORDER = "polynomial_order";
+
+template<typename PointT>
+const std::string MLSSmoothingFilter<PointT>::SEARCH_RADIUS = "search_radius";
+
+template<typename PointT>
+bool MLSSmoothingFilter<PointT>::configure(XmlRpc::XmlRpcValue value)
+{
+  std::string error;
+  if (!value.hasMember(SEARCH_RADIUS))
+    error += SEARCH_RADIUS + ", ";
+  if (!value.hasMember(POLYNOMIAL_ORDER))
+    error += POLYNOMIAL_ORDER + ", ";
+
+  if (!error.empty())
+  {
+    CONSOLE_BRIDGE_logError("Failed to find required configuration parameters: %s", error.c_str());
+    return false;
+  }
+
+  try
+  {
+    params.search_radius = static_cast<double>(value[SEARCH_RADIUS]);
+    params.polynomial_order = static_cast<int>(value[POLYNOMIAL_ORDER]);
+  }
+  catch (const XmlRpc::XmlRpcException &ex)
+  {
+    CONSOLE_BRIDGE_logError("Error configuring MLS Smoothing Filter: '%s'", ex.getMessage().c_str());
+    return false;
+  }
+
+  return true;
+}
+
+template<typename PointT>
+bool MLSSmoothingFilter<PointT>::filter(const T &input, T &output)
+{
+  // Convert the input cloud to XYZ type
+  pcl::PointCloud<pcl::PointXYZ>::Ptr cloud(new pcl::PointCloud<pcl::PointXYZ>());
+  pcl::copyPointCloud(input, *cloud);
```

Comment:
Could you create a `boost::shared_ptr` from the input with a fake deleter function as in the other plugins?

Reply:
No because the MLS smoothing filter requires an input of type `pcl::PointXYZ` which is different than the template in most cases